File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        logLine = logLine.replaceFirst("\\[\\w+]:", "");
        logLine = logLine.trim();
        return logLine;
    }

    public static String logLevel(String logLine) {
        String[] tokens = logLine.split(":");
        return tokens[0].substring(1, tokens[0].length() - 1).toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;
using System.Globalization;

static class LogLine
{
    public static string Message(string logLine)
    {
        var arr = logLine.Split(":");
        return arr[1].Trim();
    }

    public static string LogLevel(string logLine)
    {
        var parts = logLine.Split(":");
        var levelPart = parts[0];
        return RemoveBrackets(levelPart).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }

    private static string RemoveBrackets(string s)
    {
        return s.Substring(1, s.Length - 2);
    }
}
